lib/multiaddr.js

```diff
@@ -17,7 +17,10 @@ export function multiaddrToHttpUrl (addr) {
         new Error(`Cannot parse "${addr}": unsupported http path`, { cause: err }),
         { code: 'INVALID_HTTP_PATH' }
       )
-    }
+    } // Handle DNS-based direct http/https format
+  } else if (hostType.startsWith('dns') && (multiAddrParts[0] === 'http' || multiAddrParts[0] === 'https')) {
+    scheme = multiAddrParts[0]
+    rest = multiAddrParts.slice(1)
```

Comment:
Is it worth simplifying this for consistency with the style we use below?

```suggestion
    ;[scheme, ...rest] = multiAddrParts[0]
```

Reply:
Addressed in https://github.com/CheckerNetwork/spark-checker/commit/b71f5fc598ae9dd39273feacee240c53d1ce4f52